packages/net/tcp_connection.pony

```diff
@@ -810,63 +814,85 @@ actor TCPConnection
         _reading = true
 
         while _readable and not _shutdown_peer do
+          // exit if muted
           if _muted then
             _reading = false
             return
           end
 
+          // distribute the data we've already read that is in the `read_buf`
+          // and able to be distributed
+          while (_read_buf_offset >= _expect) and (_read_buf_offset > 0) do
+            // get data to be distributed and update `_read_buf_offset`
+            let data =
+              if _expect == 0 then
+                let data' = _read_buf = recover Array[U8] end
+                data'.truncate(_read_buf_offset)
```

Comment:
this is a "bug" i just fixed in lori. see SeanTAllen/lori@c61b165

Reply:
Yeah... I noticed that "bug" but was trying not to do more than a "port" of the code as part of this.